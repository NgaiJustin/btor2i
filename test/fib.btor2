; Handcraft example to compute the 50th Fib number
1 sort bitvec 32
2 one 1
3 one 1
4 add 1 2 3 ; Fib(3) = 1 + 1
5 one 1
6 add 1 4 5 ; Fib(4) = 1 + 2
7 add 1 4 6 ; Fib(5) = 2 + 3
8 add 1 6 7 ; Fib(6) = 3 + 5
9 add 1 7 8 ; Fib(7) = 5 + 8
10 add 1 8 9 ; Fib(8) = 8 + 13
11 add 1 9 10 ; Fib(9) = 13 + 21
12 add 1 10 11 ; Fib(10) = 21 + 34
13 add 1 11 12 ; Fib(11) = 34 + 55
14 add 1 12 13 ; Fib(12) = 55 + 89
15 add 1 13 14 ; Fib(13) = 89 + 144
16 add 1 14 15 ; Fib(14) = 144 + 233
17 add 1 15 16 ; Fib(15) = 233 + 377
18 add 1 16 17 ; Fib(16) = 377 + 610
19 add 1 17 18 ; Fib(17) = 610 + 987
20 add 1 18 19 ; Fib(18) = 987 + 1597
21 add 1 19 20 ; Fib(19) = 1597 + 2584
22 add 1 20 21 ; Fib(20) = 2584 + 4181
23 add 1 21 22 ; Fib(21) = 4181 + 6765
24 add 1 22 23 ; Fib(22) = 6765 + 10946
25 add 1 23 24 ; Fib(23) = 10946 + 17711
26 add 1 24 25 ; Fib(24) = 17711 + 28657
27 add 1 25 26 ; Fib(25) = 28657 + 46368
28 add 1 26 27 ; Fib(26) = 46368 + 75025
29 add 1 27 28 ; Fib(27) = 75025 + 121393
30 add 1 28 29 ; Fib(28) = 121393 + 196418
31 add 1 29 30 ; Fib(29) = 196418 + 317811
32 add 1 30 31 ; Fib(30) = 317811 + 514229
33 add 1 31 32 ; Fib(31) = 514229 + 832040
34 add 1 32 33 ; Fib(32) = 832040 + 1346269
35 add 1 33 34 ; Fib(33) = 1346269 + 2178309
36 add 1 34 35 ; Fib(34) = 2178309 + 3524578
37 add 1 35 36 ; Fib(35) = 3524578 + 5702887
38 add 1 36 37 ; Fib(36) = 5702887 + 9227465
39 add 1 37 38 ; Fib(37) = 9227465 + 14930352
40 add 1 38 39 ; Fib(38) = 14930352 + 24157817
41 add 1 39 40 ; Fib(39) = 24157817 + 39088169
42 add 1 40 41 ; Fib(40) = 39088169 + 63245986
43 add 1 41 42 ; Fib(41) = 63245986 + 102334155
44 add 1 42 43 ; Fib(42) = 102334155 + 165580141
45 add 1 43 44 ; Fib(43) = 165580141 + 267914296
46 add 1 44 45 ; Fib(44) = 267914296 + 433494437
47 add 1 45 46 ; Fib(45) = 433494437 + 701408733
48 add 1 46 47 ; Fib(46) = 701408733 + 1134903170
49 add 1 47 48 ; Fib(47) = 1134903170 + 1836311903 = 2,971,215,073 
50 output 49 fibout
; (too big) 50 add 1 48 49 ; Fib(48) = 1836311903 + 2971215073
; (too big) 51 add 1 49 50 ; Fib(49) = 2971215073 + 4807526976
; (too big) 52 add 1 50 51 ; Fib(50) = 4807526976 + 7778742049



