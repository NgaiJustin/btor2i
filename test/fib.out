Arguments:

Environment:
0: _
1: _
2: BitVector { bits: [11111111111111111111111111111111 ], width: 32 }
3: BitVector { bits: [11111111111111111111111111111111 ], width: 32 }
4: BitVector { bits: [11111111111111111111111111111110 ], width: 32 }
5: BitVector { bits: [11111111111111111111111111111111 ], width: 32 }
6: BitVector { bits: [11111111111111111111111111111101 ], width: 32 }
7: BitVector { bits: [11111111111111111111111111111011 ], width: 32 }
8: BitVector { bits: [11111111111111111111111111111000 ], width: 32 }
9: BitVector { bits: [11111111111111111111111111110011 ], width: 32 }
10: BitVector { bits: [11111111111111111111111111101011 ], width: 32 }
11: BitVector { bits: [11111111111111111111111111011110 ], width: 32 }
12: BitVector { bits: [11111111111111111111111111001001 ], width: 32 }
13: BitVector { bits: [11111111111111111111111110100111 ], width: 32 }
14: BitVector { bits: [11111111111111111111111101110000 ], width: 32 }
15: BitVector { bits: [11111111111111111111111100010111 ], width: 32 }
16: BitVector { bits: [11111111111111111111111010000111 ], width: 32 }
17: BitVector { bits: [11111111111111111111110110011110 ], width: 32 }
18: BitVector { bits: [11111111111111111111110000100101 ], width: 32 }
19: BitVector { bits: [11111111111111111111100111000011 ], width: 32 }
20: BitVector { bits: [11111111111111111111010111101000 ], width: 32 }
21: BitVector { bits: [11111111111111111110111110101011 ], width: 32 }
22: BitVector { bits: [11111111111111111110010110010011 ], width: 32 }
23: BitVector { bits: [11111111111111111101010100111110 ], width: 32 }
24: BitVector { bits: [11111111111111111011101011010001 ], width: 32 }
25: BitVector { bits: [11111111111111111001000000001111 ], width: 32 }
26: BitVector { bits: [11111111111111110100101011100000 ], width: 32 }
27: BitVector { bits: [11111111111111101101101011101111 ], width: 32 }
28: BitVector { bits: [11111111111111100010010111001111 ], width: 32 }
29: BitVector { bits: [11111111111111010000000010111110 ], width: 32 }
30: BitVector { bits: [11111111111110110010011010001101 ], width: 32 }
31: BitVector { bits: [11111111111110000010011101001011 ], width: 32 }
32: BitVector { bits: [11111111111100110100110111011000 ], width: 32 }
33: BitVector { bits: [11111111111010110111010100100011 ], width: 32 }
34: BitVector { bits: [11111111110111101100001011111011 ], width: 32 }
35: BitVector { bits: [11111111110010100011100000011110 ], width: 32 }
36: BitVector { bits: [11111111101010001111101100011001 ], width: 32 }
37: BitVector { bits: [11111111011100110011001100110111 ], width: 32 }
38: BitVector { bits: [11111111000111000010111001010000 ], width: 32 }
39: BitVector { bits: [11111110100011110110000110000111 ], width: 32 }
40: BitVector { bits: [11111101101010111000111111010111 ], width: 32 }
41: BitVector { bits: [11111100001110101111000101011110 ], width: 32 }
42: BitVector { bits: [11111001111001101000000100110101 ], width: 32 }
43: BitVector { bits: [11110110001000010111001010010011 ], width: 32 }
44: BitVector { bits: [11110000000001111111001111001000 ], width: 32 }
45: BitVector { bits: [11100110001010010110011001011011 ], width: 32 }
46: BitVector { bits: [11010110001100010101101000100011 ], width: 32 }
47: BitVector { bits: [10111100010110101100000001111110 ], width: 32 }
48: BitVector { bits: [10010010100011000001101010100001 ], width: 32 }
49: BitVector { bits: [1001110111001101101101100011111 ], width: 32 }
50: _
51: _
52: _
53: _
54: _
55: _
56: _

Output:
fibout: BitVector { bits: [1001110111001101101101100011111 ], width: 32 }
